OpenRA.Mods.Common/Traits/Render/WithRandomIntervalOverlay.cs

```diff
@@ -0,0 +1,111 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2017 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System;
+using System.Collections.Generic;
+using OpenRA.Graphics;
+using OpenRA.Mods.Common.Graphics;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits.Render
+{
+	[Desc("Renders a looping animation over units and buildings between random intervals.")]
+	public class WithRandomIntervalOverlayInfo : PausableConditionalTraitInfo
```

Comment:
This would be better implemented as a new delay field on `WithIdleOverlay`.  See also the discussion in #13737 which covers the same thing but for the D2K thumper.

Reply:
Did that at first, but it would mean all idle overlays would tick so I abandoned that.